pkg/upgrade/handle_upgrade.go

```diff
@@ -59,7 +59,7 @@ func (ctl *Controller) handlePlans(ctx context.Context) error {
 	)
 
 	// process plan events by creating jobs to apply the plan
-	upgradectlv1.RegisterPlanGeneratingHandler(ctx, plans, ctl.apply.WithCacheTypes(jobs, nodes, secrets).WithNoDelete(), "", ctl.Name,
+	upgradectlv1.RegisterPlanGeneratingHandler(ctx, plans, ctl.apply.WithCacheTypes(nodes, secrets).WithGVK(jobs.GroupVersionKind()).WithDynamicLookup().WithNoDelete(), "", ctl.Name,
```

Comment:
So this is more correct but it didn't get us anything, is that right @MonzElmasry?

Reply:
yes!